packages/cli/src/utils/generate.ts

```diff
@@ -0,0 +1,160 @@
+import fse from 'fs-extra'
+import fs from 'fs'
+import deepmerge from 'deepmerge'
+
+import {
+  CMSCoreDir,
+  CMSCustomDir,
+  CMSTmpDir,
+  coreDir,
+  customizationsDir,
+  customSrcDir,
+  repoDir,
+  storeConfigCoreFileDir,
+  storeConfigFileDir,
+  storeConfigTmpFileDir,
+  themeCustomizationsFileDir,
+  themesFileDir,
+  tmpDir,
+  tmpFolderName,
+} from './directory'
+
+interface GenerateOptions {
+  setup?: boolean
+}
+
+const ignoredDirs = ['node_modules']
+
+async function importStoreConfig() {
+  const { default: storeConfig } = await import(`${repoDir}/store.config.js`)
+  return storeConfig
+}
+
+function createTmpFolder() {
+  try {
+    fse.mkdirsSync(tmpDir)
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log(`Temporary folder ${tmpFolderName} created`)
+  }
+}
+
+function copyCoreFiles() {
+  try {
+    fse.copySync(coreDir, tmpDir, {
+      filter(src) {
+        const splittedSrc = src.split('/')
+        return ignoredDirs.indexOf(splittedSrc[splittedSrc.length - 1]) === -1
+      },
+    })
+  } catch (e) {
+    console.error(e)
+  } finally {
+    console.log(`Core files copied`)
+  }
+}
+
+function copyCustomSrcToCustomizations() {
+  try {
+    fse.copySync(customSrcDir, customizationsDir)
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log('Copied custom files')
+  }
+}
+
+async function copyTheme() {
+  const storeConfig = await importStoreConfig()
+
+  try {
+    fs.copyFileSync(
+      `${themesFileDir}/${storeConfig.theme}.scss`,
+      themeCustomizationsFileDir
+    )
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log('Copied custom styles')
+  }
+}
+
+function mergeCMSFile(fileName: string) {
+  const coreContentTypes = fs.readFileSync(`${CMSCoreDir}/${fileName}`, 'utf8')
+  const customContentTypes = fs.readFileSync(
+    `${CMSCustomDir}/${fileName}`,
+    'utf8'
+  )
+  const coreContentTypesJSON = JSON.parse(coreContentTypes)
+  const customContentTypesJSON = JSON.parse(customContentTypes)
+
+  const mergeContentTypes = [...coreContentTypesJSON, ...customContentTypesJSON]
+
+  try {
+    fs.writeFileSync(
+      `${CMSTmpDir}/${fileName}`,
+      JSON.stringify(mergeContentTypes)
+    )
+  } catch (err) {
+    console.error(err)
+  } finally {
+    console.log(`CMS file ${fileName} created`)
+  }
+}
+
+function generateStoreConfigFile(content: any) {
+  return `module.exports = ${JSON.stringify(content)}\n`
+}
```

Comment:
We can type this to match the current `store.config.js` file for now, could be a good move to make sure we keep the same API, and even if we're changing it, changes would be more intentional

Reply:
I'm not 100% if I got your comment right, but I think you are talking about typing the `content` param, right? If so, I see your point, but would this add any value to the CLI? I see it as a good but useless move since typescript would not guarantee the types were right anyway.

Reply:
Yes! I was referring to the `content`.

I see your point hahahah maybe TypeScript would not be that useful here, but what would certainly be of value is to have some sort of "validation" step to run on all configuration files that are generated from user input, right?

Reply:
I agree with you! I think we should have this. I think we'll have to work on typing this on the js side (just like nextjs does) and trimming available config options as well as moving config options to other files (the case with lighthouse). I think we should discuss this API and once settled, we can implement this validation step. What do you think?